refactor: 광고 감지 여부에 따른 whisper 서버 연결

faber-h opened the pull request:
### ✨ 이슈 번호

- #73 

### 📌 설명

- 광고 감지(isAdDetect) 여부에 따라 Whisper 서버로 요청을 보내는 로직을 추가를 구현한 Pull Request입니다.
- ChannelPlayer 컴포넌트에서 isAdDetect를 전달해주도록 변경했습니다.
- MiniPlayer 컴포넌트를 ChannelPlayer와 동일한 로직 흐름으로 변경했습니다.

### 📃 작업 사항

  - [x] getChannelInfo API에 광고 감지 여부(isAdDetect)를 query param으로 전달하도록 수정
  - [x] controlStreamingPlayback 함수에서 isAdDetect 여부에 따라 Whisper 서버 요청 여부 분기 처리
  - [x] ChannelPlayer controlStreamingPlayback 호출 시 isAdDetect 전달 로직 추가
    - [x] MiniPlayer 컴포넌트를 ChannelPlayer와 동일한 로직 흐름으로 변경

### 💡 참고 사항

- 백엔드와 함께 진행된 PR입니다.
- [백엔드 광고 감지 여부에 따른 whisper 서버 연결 PR](https://github.com/Radio-Premium/RadioPremium-BE/pull/53)

### 💭 리뷰 사항 반영 

- [x] 역할에 맞게 변수명 명확화
- [x] 구조 분해 할당을 사용 find 로직 간결화
- [x] useChannelPlayback로 ChannelPlayer, MiniPlayer 중복 로직 추출

### ✅ Pull Request 체크 사항

- [x] 가장 최신 브랜치를 pull 하였습니다.
- [x] base 브랜치명을 확인하였습니다.
- [x] 코드 컨벤션을 모두 확인하였습니다.
- [x] 브랜치명을 확인하였습니다.

faber-h committed:
feat: getChannelInfo API에 광고 감지 여부 파라미터 추가

- getChannelInfo(channelId, isAdDetect) 형태로 변경
- 광고 감지 여부에 따라 백엔드 whisper 연동 유무 결정

faber-h committed:
refactor: controlStreamingPlayback에 광고 감지 옵션 전달 기능 추가

faber-h committed:
refactor: ChannelPlayer에서 광고 감지 설정값 전달하도록 수정

faber-h committed:
refactor: MiniPlayer에 광고 감지 설정값 반영

- ChannelPlayer와 동일한 로직 흐름으로 변경
- useUserStore에서 isAdDetect 가져와서 재생 함수에 전달

faber-h committed:
refactor: 상태 및 로직 선언 순서 정리

devjaeyoon commented on src/components/MiniPlayer.jsx:
```diff
@@ -1,28 +1,36 @@
+  const { settings } = useUserStore();
+  const isAdDetect = settings[SETTING_TYPES.AD_DETECT];
+
+  const channel = radioChannelList.find(
```
`radioChannelList`에서 `selectedChannelId`에 해당하는 채널을 찾는거라 `selectedChannel` 변수명은 어떠신가요?

faber-h replied:
좋은 의견 감사해요!
해당 변수는 실제로 `selectedChannelId`에 해당하는 채널을 찾는 역할이라
`selectedChannel`이라는 이름이 역할을 더 잘 드러내는 것 같네요.
변수명을 `channel` → `selectedChannel`로 변경했습니다 :)

devjaeyoon commented on src/components/MiniPlayer.jsx:
```diff
@@ -1,28 +1,36 @@
+  const isAdDetect = settings[SETTING_TYPES.AD_DETECT];
+
+  const channel = radioChannelList.find(
+    (channel) => channel.id === selectedChannelId
```
```suggestion
    ({ id }) => id === selectedChannelId
```
구조 분해 할당으로 축약할 수 있을 것 같아요!

faber-h replied:
좋은 피드백 감사해요!
구조 분해를 통해 `({ id }) => id === selectedChannelId`로 작성하면 코드가 더 간결해지네요!
수정하여 반영했습니다 :)

devjaeyoon commented on src/components/MiniPlayer.jsx:
```diff
@@ -1,28 +1,36 @@
 
-const MiniPlayer = ({ thumbnail, channelName, closePlayer }) => {
-  const [isPlaying, setIsPlaying] = useState(false);
+const MiniPlayer = ({ closePlayer }) => {
```
지금 `MiniPlayer`와 `ChannelPlayer`에 채널 정보를 가져오는 로직이 비슷한 것 같은데 `useChannelData`같은 훅을 만들어 하나로 합칠 수 있는 방법은 없을까요?

faber-h replied:
좋은 리뷰 감사해요~
채널 정보를 가져오는 로직을 포함한 중복된 로직들을 처리하기 위해서 `useChannelPlayback` 훅을 만들었습니다 😀

faber-h committed:
refactor: 역할에 맞게 변수명 명확화

faber-h committed:
refactor: 구조 분해 할당을 사용 find 로직 간결화

faber-h committed:
refactor: useChannelPlayback로 중복 로직 추출

- ChannelPlayer, MiniPlayer 중복 로직
- useChannelStore()에서 동일한 상태 추출
- useUserStore()에서 settings 추출 후 isAdDetect 계산
- radioChannelList.find(...)로 현재 선택된 채널 정보 조회
- controlStreamingPlayback() 호출 로직 동일

d0-neii commented on src/pages/ChannelPlayer.jsx:
```diff
@@ -1,34 +1,19 @@
-  const videoId = useRef(null);
-
-  const { name, logoUrl } = channel;
+  const { videoId, selectedChannel, isPlaying, handlePlayPause } =
```
`useChannelPlayback`으로 관련 상태와 핸들러가 잘 추출되어 컴포넌트가 훨씬 간결해진 것 같습니다!
수고하셨습니다🤩

faber-h replied:
감사합니다 ☺️

faber-h committed:
Merge branch 'dev' into refactor/#73

Eun0713 reviewed (approved):
LGTM🌻

devjaeyoon reviewed (approved):
고생하셨습니다! LGTM!
